lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -380,6 +380,17 @@ public function getListTableColumnsSQL($table, $database = null)
                'FROM information_schema.COLUMNS WHERE TABLE_SCHEMA = ' . $database . ' AND TABLE_NAME = ' . $table;
     }
 
+    public function getListTablesMetadataSQL($database = null)
+    {
+        $database = $database === null
+            ? 'DATABASE()'
+            : $this->quoteStringLiteral($database);
+
+        return 'SELECT TABLE_NAME, ENGINE, AUTO_INCREMENT, TABLE_COLLATION, TABLE_COMMENT, CREATE_OPTIONS'
```

Comment:
Actually, I started writing this using an HEREDOC but I found the indentation very bad (It could be improved once php 7.3+ will be required) and I do not know whether new lines where accepted in code. Do you prefer `sprintf`` or variable interpolation? I guess interpolation is faster (but I can be wrong).

Reply:
It's not really about performance but more about readability. With concatenation, in order to copy the query somewhere, you'll need to remove all PHP bits from the code first. Furthermore, with `sprintf()`, it's easier to see the logic behind the query. Please use `sprintf()`.

There's nothing wrong about HEREDOC indentation in PHP prior to 7.3. In most cases, HEREDOCs are used for "code in code" which doesn't have to be indented the same. You can  think of it as of a boxy in typography:
```php
if (true) {
    if (true) {
        return sprintf(
            <<<'SQL'
SELECT foo
FROM bar
WHERE %s
SQL
            ,
            $value
        );
    }
}
```